### ISSUE TITLE
How to upgrade from 2.13 to 3.x?

### ISSUE THREAD

**Issue by bikeymouse** [xirixiz/dsmr-reader-docker#90]
Hi,

I'm running the latest available V2 Docker image for the RPI, which is arm32v6-v2.13.0. 
Now I would like to upgrade to the V3 branch.

According to [the documentation](https://dsmr-reader.readthedocs.io/en/v3/faq/v3_upgrade.html) I should first upgrade to V2.15. Unfortunately there is no image for that. 

I tried to directly doing the upgrade by starting the V3.1 image but that fails, which seems to be because there where some database changes between V2.12 and V2.15.

I also tried to run ./deploy.sh to do the upgrade inside the V2.12 container, but that also fails ("FAILED to switch to 'dsmrreader' VirtualEnv (is it installed?)).

So question is: what can I do to upgrade to the latest V3.x docker image?

**Comment by xirixiz** [comment]
I’ll create a 2.15 docker image 👍🏻. I’ll do that this morning.

**Comment by xirixiz** [comment]
Added!

**Comment by bikeymouse** [comment]
Great thanks!

However, I just installed it, and it seems that the jump from V2.12 to V2.15 is too big.

DSMR won't start, and in the postgress logging I see: 
_`ERROR:  column dsmr_datalogger_dataloggersettings.process_sleep does not exist at character 153`_

That is not surprising as I see in the Changelog that in 2.14 (and maybe 2.13) there where database changes: " Move custom configuration in settings.py to database".

So probably I have to upgrade first to 2.14 (or even to 2.13). Can you make images for those too? 
Or think how to make upgrading within the running container possible?

**Comment by mcoonen** [comment]
Thanks xirixiz for the latest docker images. I succesfully upgraded.

@bikeymouse, I did not need v2.14. What worked for me was:
1. Upgrade v2.11.3 --> v2.13.0
2. Upgrade v2.13.0 --> v2.15.0
3. Upgrade v2.15.0 --> v3.1.0

All steps performed by simply pulling the docker image with specified version tag, starting the docker container (you'll see the migrations being active in log), creating a full backup, stopping docker container. And then repeat the steps for the next upgrade.

**Comment by bikeymouse** [comment]
Mmm, that is strange I just tried to upgrade from V2.13 to V2.15 (mistyped this in my message above), but that failed, basically on the database changes that where made in V2.14

**Comment by bikeymouse** [comment]
Ok, I have found the problem. 

For one reason or another, the migrate script that runs when starting the container had problems with the value for the back-end sleep that I had set by using the environment variables in my docker-compose.yml, which worked fine until now. I got an error with something about a decimal value.

I had
      - DSMR_BACKEND_SLEEP=120
      - DSMR_DATALOGGER_SLEEP=60

I now have removed these from the docker-compose which is fine because they can now be managed from inside the Django settings page in the DSMR application.

Just posting this, for anybody else who may encounter the same problem.

Thanks again for the help!
